### PR TITLE
Fixes #3129 - sp_BlitzFirst in Azure SQL Managed Instances

### LINKED ISSUE #3129: sp_BlitzFirst Check 44 in Azure Managed Instance fails with error
**Version of the script**
Version '8.10', VersionDate '20220718'

**What is the current behavior?**
Runs through until check 44 then reports "Database '?' does not exist. Make sure the name is entered correctly."

**If the current behavior is a bug, please provide the steps to reproduce.**
execute sp_BlitzFirst @expertMode = 1 in an Azure MI

**What is the expected behavior?**
No error

**Which versions of SQL Server and which OS are affected by this issue? Did this work in previous versions of our procedures?**
New user, so no previous versions tested. I believe Azure MI is in constant state of revision but I think I have a fix:

Unless I am misunderstanding the purpose of the IF...ELSE checks for 'SQL Azure' prior to the population of @StringToExecute and similarly the use of sp_MSForeachdb, shouldn't line 2478 be updated to not include 'USE [?]; '? as it's being appended if required first?

Thank you:-)

### PR DESCRIPTION
**Pull request by dmonlineuk**
Retains checks for Azure SQL DB and tested to work there, but now enables it to work for Azure SQL Managed Instances as (I believe is) intended - also tested.

TLDR - looked for SERVERPROPERTY('Edition') =/<> 'Azure SQL' and updated to SERVERPROPERTY('EngineEdition') =/<> 5

### CODE REVIEW

**Review by BrentOzar — APPROVED** [review]
Thanks for the pull request! Looks good, merging into the dev branch. Will be in the next release with credit to you in the release notes.
